programs/intent-transfer/src/lib.rs

```diff
@@ -28,6 +36,368 @@ pub mod intent_transfer {
         ctx.accounts
             .verify_and_send(&[&[INTENT_TRANSFER_SEED, &[ctx.bumps.intent_transfer_setter]]])
     }
+
+    #[instruction(discriminator = [1])]
+    pub fn bridge_ntt_tokens<'info>(
+        ctx: Context<'_, '_, '_, 'info, BridgeNttTokens<'info>>,
+        args: BridgeNttTokensArgs,
+    ) -> Result<()> {
+        ctx.accounts.verify_and_initiate_bridge(
+            &[&[INTENT_TRANSFER_SEED, &[ctx.bumps.intent_transfer_setter]]],
+            args,
+        )
+    }
+}
+
+// TODO: we should do some parsing of the relay_instructions and/or exec_amounts arg(s)
+// in order to ensure the signed intent message does precisely what the user expects
+#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
+pub struct BridgeNttTokensArgs {
+    pub exec_amount: u64,
+    pub signed_quote_bytes: Vec<u8>,
+    pub relay_instructions: Vec<u8>,
+}
+
+#[derive(Accounts)]
+pub struct Ntt<'info> {
+    /// CHECK: Clock sysvar
+    pub clock: Sysvar<'info, Clock>,
+
+    /// CHECK: Rent sysvar
+    pub rent: Sysvar<'info, Rent>,
+
+    /// CHECK: checked in NTT manager program
+    pub ntt_manager: UncheckedAccount<'info>,
```

Comment:
do you need to check the address of this so it's not possible to CPI to an arbritary program?

Alternatively, `cpi::transfer_burn` can become `cpi::transfer_burn(ctx, args)` and hardcode program_id to the correct address of the program, and the CPI will fail if ntt_manager is not what it is.

Reply:
ah this is this way bc the NTT manager program is specific to the mint, and there are different NTT managers for different mints. The NTT manager program has a check to make sure that the right mint is passed in